src/js/test/unit/package-manager.test.ts

```diff
@@ -1,59 +1,93 @@
-import { PackageManager } from "../../load-package.ts";
-import { PackageManagerAPI, PackageManagerModule } from "../../types.ts";
+import * as chai from "chai";
+import sinon from "sinon";
+import { PackageManager, toStringArray } from "../../load-package.ts";
+import { genMockAPI, genMockModule } from "./test-helper.ts";
 
 describe("PackageManager", () => {
-  // TODO: add more unittests
   it("should initialize with API and Module", () => {
-    const mockApi: PackageManagerAPI = {
-      importlib: {
-        invalidate_caches: () => {},
-      },
-      package_loader: {
-        get_install_dir: () => "",
-        init_loaded_packages: () => {},
-        unpack_buffer: {
-          callKwargs: () => {},
-        },
-      },
-      config: {
-        indexURL: "",
-        packageCacheDir: "",
-      },
-      lockfile_packages: {},
-      bootstrapFinalizedPromise: Promise.resolve(),
-      sitepackages: "",
-      defaultLdLibraryPath: [],
+    const mockApi = genMockAPI();
+    const mockMod = genMockModule();
+    const _ = new PackageManager(mockApi, mockMod);
+  });
+});
+
+describe("logStdout and logStderr", () => {
+  it("Should use console.log and console.error if no logger is provided", () => {
+    const mockApi = genMockAPI();
+    const mockMod = genMockModule();
+
+    const pm = new PackageManager(mockApi, mockMod);
+
+    const logSpy = sinon.spy(pm, "stdout");
+    const errorSpy = sinon.spy(pm, "stderr");
+
+    pm.logStdout("stdout");
+    pm.logStderr("stderr");
+
+    chai.assert.isTrue(logSpy.calledOnce);
+    chai.assert.isTrue(errorSpy.calledOnce);
+    chai.assert.isTrue(logSpy.calledWith("stdout"));
+    chai.assert.isTrue(errorSpy.calledWith("stderr"));
+
+    logSpy.restore();
+    errorSpy.restore();
+  });
+
+  it("Should use the provided logger if one is provided", () => {
+    const mockApi = genMockAPI();
+    const mockMod = genMockModule();
+
+    const pm = new PackageManager(mockApi, mockMod);
+
+    const stdoutLogger = sinon.spy();
+    const stderrLogger = sinon.spy();
+
+    pm.logStdout("stdout", stdoutLogger);
+    pm.logStderr("stderr", stderrLogger);
+
+    chai.assert.isTrue(stdoutLogger.calledOnce);
+    chai.assert.isTrue(stderrLogger.calledOnce);
+    chai.assert.isTrue(stdoutLogger.calledWith("stdout"));
+    chai.assert.isTrue(stderrLogger.calledWith("stderr"));
+  });
+});
+
+describe("toStringArray", () => {
+  it("Should convert string to array of strings", () => {
+    chai.assert.deepEqual(toStringArray("hello"), ["hello"]);
+  });
+
+  it("Should return the array if it is already an array", () => {
+    chai.assert.deepEqual(toStringArray(["hello", "world"]), [
+      "hello",
+      "world",
+    ]);
+  });
+
+  it("Should convert PyProxy to array of strings", () => {
+    // TODO: use real PyProxy
+    const pyProxyMock = {
+      toJs: () => "hello",
```

Comment:
While you're at it, maybe add another where `toJs()` returns a list? This is the more realistic case, I don't know if `toJs` would ever return a string in real life.

Reply:
I changed the test to return the list. Thanks for the suggestion.